torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -565,13 +602,33 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
         nullptr));
   }
 
-  if (measure_kernel_time_) {
+  if (measure_kernel_time_ ||
+      isDebugDumpEnabled(DebugDumpOption::EffectiveBandwidth)) {
     cudaEventRecord(finish_event);
     cudaEventSynchronize(start_event);
     cudaEventSynchronize(finish_event);
     cudaEventElapsedTime(&kernel_time_ms_, start_event, finish_event);
     cudaEventDestroy(start_event);
     cudaEventDestroy(finish_event);
+
+    if (isDebugDumpEnabled(DebugDumpOption::EffectiveBandwidth)) {
+      size_t bytes = 0;
+      // Figure how many bytes are inputs, outputs, and temporary buffers
+      for (auto input : inputs) {
```

Comment:
Should this depend on the number of uses for each input?
e.g., A non-persistent kernel could read an input multiple times.

Reply:
It's not perfect, both seem to have a limitation. I think this gives the best idea of what the best case scenario could be with a quick glance. Will still be up to us to do more indepth analysis, though maybe we could build smarter systems when we have a more complete version of persistence which evaluates if persistence could even be used.